apps/api/src/report/entity/hired-applicant-milestone.entity.ts

```diff
@@ -0,0 +1,389 @@
+import { ViewColumn, ViewEntity } from 'typeorm';
+import { HiredApplicantEntity } from './hired-applicant.entity';
+
+// the expression is given by reference
+// because we don't sync schema and the view table should have been created by a migration
+@ViewEntity({
+  name: 'hired_applicant_milestone',
+  expression: `
+    SELECT
+      hwa.id,
+        hwa.registered_at,
+        hwa.hired_at,
+        hwa.withdrew_at,
+        hwa.ha,
+        (
+          SELECT
+            LEAST(min(asa.start_date), hwa.registered_at) AS "least"
+          FROM
+            ien_applicant_status_audit asa
+          WHERE
+            asa.applicant_id = hwa.id
+        ) AS milestone_start_date,
+        
+        (
+          SELECT
+            min(asa.start_date) AS min
+          FROM
+            ien_applicant_status_audit asa
+          WHERE
+            asa.applicant_id = hwa.id
+            AND (asa.status_id = ANY (ARRAY['e68f902c-8440-4a9c-a05f-2765301de800'::uuid,
+            '897156c7-958c-4ec0-879a-ed4943af7b72'::uuid,
+            '20268c6e-145e-48cd-889a-2346985db957'::uuid]))
```

Comment:
do these all need to be hardcoded?

Reply:
It's generated from the view table and I added it as a comment for reference.
To map IDs dynamically, we need to use a function with a param of `datasource` and write typeorm query. I just didn't want to bother the codes that won't be run.